iceoryx2-bb/posix/tests/socket_pair_tests.rs

```diff
@@ -0,0 +1,325 @@
+    let mut number_of_data_sent = 0;
+
+    loop {
+        let result = sut_lhs.try_send(&send_data).unwrap();
+        if result != send_data.len() {
+            break;
+        }
+        number_of_data_sent += 1;
+    }
+
+    let barrier = Barrier::new(2);
+    let counter = IoxAtomicUsize::new(0);
+    std::thread::scope(|s| {
+        s.spawn(|| {
+            barrier.wait();
+            let result = sut_lhs.timed_send(&send_data, TIMEOUT * 100).unwrap();
+            counter.store(1, Ordering::Relaxed);
+            assert_that!(result, eq send_data.len());
+        });
+
+        let mut receive_buffer = vec![];
+        receive_buffer.resize(number_of_data_sent, 0);
+
+        barrier.wait();
+        std::thread::sleep(TIMEOUT);
+        assert_that!(counter.load(Ordering::Relaxed), eq 0);
+        let result = sut_rhs.try_receive(&mut receive_buffer).unwrap();
+        assert_that!(result, eq number_of_data_sent);
+        for byte in receive_buffer {
+            assert_that!(byte, eq b'Q');
+        }
+    });
+}
+
+#[test]
+fn blocking_send_blocks_until_message_buffer_is_free_again() {
+    let _watchdog = Watchdog::new();
+
+    let (sut_lhs, sut_rhs) = StreamingSocket::create_pair().unwrap();
+
+    let send_data = Vec::from(b"X");
+    let mut number_of_data_sent = 0;
+
+    loop {
+        let result = sut_lhs.try_send(&send_data).unwrap();
+        if result != send_data.len() {
+            break;
+        }
+        number_of_data_sent += 1;
+    }
+
+    let barrier = Barrier::new(2);
+    let counter = IoxAtomicUsize::new(0);
+    std::thread::scope(|s| {
+        s.spawn(|| {
+            barrier.wait();
+            let result = sut_lhs.blocking_send(&send_data).unwrap();
+            counter.store(1, Ordering::Relaxed);
+            assert_that!(result, eq send_data.len());
+        });
+
+        let mut receive_buffer = vec![];
+        receive_buffer.resize(number_of_data_sent, 0);
+
+        barrier.wait();
+        std::thread::sleep(TIMEOUT);
+        assert_that!(counter.load(Ordering::Relaxed), eq 0);
+        let result = sut_rhs.try_receive(&mut receive_buffer).unwrap();
+        assert_that!(result, eq number_of_data_sent);
+
+        for byte in receive_buffer {
+            assert_that!(byte, eq b'X');
+        }
+    });
+}
+
+#[test]
+fn peeking_message_does_not_remove_message() {
+    let _watchdog = Watchdog::new();
+
+    let (sut_lhs, sut_rhs) = StreamingSocket::create_pair().unwrap();
+
+    let send_data = Vec::from(b"get schwifty!");
+
+    let result = sut_lhs.try_send(&send_data);
+    assert_that!(result, is_ok);
+    assert_that!(result.unwrap(), eq send_data.len());
+
+    for _ in 0..5 {
+        let mut received_data = vec![];
+        received_data.resize(send_data.len(), 0);
+        let result = sut_rhs.peek(&mut received_data);
+        assert_that!(result, is_ok);
+        assert_that!(result.unwrap(), eq send_data.len());
+        assert_that!(send_data, eq received_data);
+    }
```

Comment:
Please also read the message at the end to ensure that peeking does not break receiving.